docs/content/en/docs/concepts/kube_context/_index.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Kube-context activation"
+linkTitle: "Kube-context activation"
+weight: 80
+---
+
+This page discusses how Skaffold selects the kube-context.
+
+
+When interacting with a kubernetes cluster, Skaffold does so via a kube-context.
+Thus, the selected kube-context determines the kubernetes cluster, the kubernetes user, and the default namespace.
+By default, Skaffold uses the _current_ kube-context from your kube-config file.
+
+To override this default, Skaffold offers two options:
+
+1. Via the `kube-context` flag
+
+    ```bash
+    skaffold dev --kube-context <myrepo>
+    ```
+
+1. Via the `deploy.kubeContext` option in `skaffold.yaml`
+
+    ```yaml
+    deploy:
+      kubeContext: minikube
+    ```
+
+When both are given, the CLI flag always takes precedence.
+
+### Kube-context activation and Skaffold profiles
+
+The kube-context has a double role for Skaffold profiles:
+
+1. Profiles may be auto-activated by a given kube-context.
+
+1. It is possible to change the kube-context through a `deploy.kubeContext` option in a Skaffold profile.
+
+For clarity, Skaffold does the whole profile activation with the original kube-context.
+In addition, when a profile is auto-activated by a matching kube-context, the resulting kube-context must remain unchanged.
+This rule prevents profile-specific settings for one context to be deployed into a different context.
+
+
+For example, given the following profiles:
+```yaml
+profiles:
+  - name: profile-1
+    deploy:
+      kubeContext: docker-for-desktop
+
+  - name: profile-2
+    activation:
+      - kubeContext: minikube
+```
+
+It is illegal to activate both profiles which happens when
```

Comment:
you could re-word this as
```suggestion
Activating multiple `kube-contexts` will result in error.   
```
You could also mention the error we print here.

Reply:
That wording sounds as if it is illegal to have multiple differing `profiles.deploy.kubeContext` configurations. However that is not the case.
What I am trying to say is that it is illegal to have automatic profile activations via a matching `kube-context` AND the effective kube-context changes.

I'll try to make that point clearer.